src/main/scala/com/campudus/tableaux/database/model/TableauxModel.scala

```diff
@@ -216,70 +305,81 @@ class TableauxModel(override protected[this] val connection: DatabaseConnection)
 
   private def mapRawRows(table: Table, columns: Seq[ColumnType[_]], rawRows: Seq[(RowId, Seq[Any])]): Future[Seq[Row]] = {
 
-    def mapNullValueForLinkToConcat(concatColumn: ConcatColumn, array: JsonArray): Future[List[Any]] = {
+    def mapNullValueForLinkToConcat(concatColumn: ConcatColumn, array: JsonArray): Future[List[JsonObject]] = {
       import scala.collection.JavaConverters._
 
       // Iterate over each linked row and
       // replace json's value with ConcatColumn value
-      Future.sequence(array.asScala.map({
-        case obj: JsonObject =>
-          // ConcatColumn's value is always a
-          // json array with the linked row ids
-          val rowId = obj.getLong("id").longValue()
-          retrieveCell(concatColumn, rowId).map(cell => Json.obj("id" -> rowId, "value" -> cell.value))
-      }).toList)
+      array.asScala.foldLeft(Future.successful(List.empty[JsonObject])) {
+        case (futureList, obj: JsonObject) =>
+          futureList.flatMap({
+            case list =>
+              // ConcatColumn's value is always a
+              // json array with the linked row ids
+              val rowId = obj.getLong("id").longValue()
+              retrieveCell(concatColumn, rowId)
+                .map(cell => Json.obj("id" -> rowId, "value" -> cell.value))
+                .map(json => list ++ List(json))
+          })
+      }
     }
 
     // TODO potential performance problem: foreach row every attachment column is mapped and attachments will be fetched!
-    val mergedRows = rawRows map {
-      case (rowId, rawValues) =>
+    val mergedRowsFuture = rawRows.foldLeft(Future.successful(List.empty[(RowId, Seq[Any])])) {
```

Comment:
Because it results in timeout exceptions and getConnection deadlocks if done concurrently. The map function does queries itself.